### PR TITLE
Add explicit get/set_datadir routines to sherpa.utils.testing

### PR DESCRIPTION
**Pull request by DougBurke**
# Summary

Internal changes to how the test data directory location is set and queried, including removing direct support from SherpaTestCase. Added tests for some of this functionality, and updated several test files to remove SherpaTestCase or use the new datadir functionality.

# Details

The main aim is to make sure we can drop the need for the SherpaTestCase object when setting and getting the data-directory path. It's part of my glacially-slow attempt at removing SherpaTestCase.

### CODE REVIEW

**Comment by DougBurke** [comment]
Rebased to get GH Actions.

**Review by Marie-Terrell — APPROVED** [review]
These changes look good to me.

File: sherpa/utils/tests/test_utils_testing.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        pytest.skip("datadir is not set")
+
+    # Ideally we'd wrap this up so that a problem would not bleed into
+    # other tests, but this would be a fairly serious problem so do not
+    # bother at this time.
+    #
+    # Assume the current directory is a reasonable directory to use.
+    #
+    npath = os.getcwd()
+    assert opath != npath  # just a sanety check
```
Comment (wmclaugh) [inline sherpa/utils/tests/test_utils_testing.py:86]: sanety -> sanity

File: sherpa/astro/ui/tests/test_astro_ui_io.py [hunk 2]
```diff
@@ -22,90 +22,83 @@
...
+    validate_pha(idval)
 
-        self.validate_pha(idval)
+    # TODO: does this indicate that the file name, as read in,
+    #       should have the .gz added to it to match the data
+    #       read in, or left as is?
+    pha = ui.get_data(idval)
+    bpha = ui.get_bkg(idval, bkg_id=1)
+    assert pha.name == bpha.name + '.gz'
+    assert pha.name == fname
```
Comment (wmclaugh) [inline sherpa/astro/ui/tests/test_astro_ui_io.py:87]: Is the assertion in 87 necessary after doing the validate_pha() check in 87?

Comment (DougBurke) [reply sherpa/astro/ui/tests/test_astro_ui_io.py:87]: I haven't looked at this code for a while, but I don't believe `validate_pha` explicitly checks the `name` attribute of the data object, so I believe it is technically testing something "new".

**Pull request merged**